langstream-core/src/test/java/ai/langstream/model/parser/ResourcesSpecsTest.java

```diff
@@ -140,4 +143,92 @@ private static String buildInstanceYaml() {
                     type: "none"
                 """;
     }
+
+    @Test
+    public void testConfigureResourceSpecsWithDisk() throws Exception {
+        Application applicationInstance =
+                ModelBuilder.buildApplicationInstance(
+                                Map.of(
+                                        "module.yaml",
+                                        """
+                        module: "module-1"
+                        id: "pipeline-1"
+                        resources:
+                           parallelism: 7
+                           size: 7
+                        topics:
+                          - name: "input-topic"
+                            creation-mode: create-if-not-exists
+                        pipeline:
+                          - name: "step1"
+                            type: "noop"
+                            input: "input-topic"
+                          - name: "step2"
+                            type: "noop"
+                            resources:
+                               disk:
+                                enabled: false
+                          - name: "step3"
+                            type: "noop"
+                            resources:
+                               disk:
+                                enabled: true
+                          - name: "step4"
+                            type: "noop"
+                            resources:
+                               disk:
+                                enabled: true
+                                type: "ssd"
+                          - name: "step5"
+                            type: "noop"
+                            resources:
+                               disk:
+                                enabled: true
```

Comment:
enabled ? it seems redundant

Reply:
because maybe you want to temporary disable it without removing the configuration.

it is hard to distinguish "not enabled" from nulls that we have in all the code paths that deal with disks, like merging configurations with defaults or handling backward/forward compatibility